File: 04583.py
while 1:
    s = input()
    if s == '#': break

    valid = True; res = ""
    for i in range(len(s)):
        if s[i] == 'b': res += 'd'
        elif s[i] == 'd': res += 'b'
        elif s[i] == 'p': res += 'q'
        elif s[i] == 'q': res += 'p'
        elif s[i] == 'i' or s[i] == 'o' or s[i] == 'v' or s[i] == 'w' or s[i] == 'x':
            res += s[i]
        else:
            valid = False
            res += s[i]

    if not valid:
        print("INVALID")
    else:
        for i in range(len(res) - 1, -1, -1):
            print(res[i], end='')
        print()

File: 04583.cpp
#include <iostream>
using namespace std;

int main() {
	while (1) {
		string s; cin >> s;
		if (s == "#") break;

		bool valid = true;
		for (int i = 0; i < s.length(); i++) {
			if (s[i] == 'b') s[i] = 'd';
			else if (s[i] == 'd') s[i] = 'b';
			else if (s[i] == 'p') s[i] = 'q';
			else if (s[i] == 'q') s[i] = 'p';
			else if (s[i] == 'i' || s[i] == 'o' || s[i] == 'v' || s[i] == 'w' || s[i] == 'x');
			else valid = false;
		}

		if (!valid) cout << "INVALID\n";
		else {
			for (int i = s.length() - 1; i >= 0; i--)
				cout << s[i];
			cout << endl;
		}
	}
}
